include/fmt/ostream.h

```diff
@@ -129,14 +129,22 @@ template <typename T> struct streamed_view { const T& value; };
 
 // Formats an object of type T that has an overloaded ostream operator<<.
 template <typename Char>
-struct basic_ostream_formatter : formatter<basic_string_view<Char>, Char> {
+struct basic_ostream_formatter {
+private:
+  formatter<basic_string_view<Char>, Char> underlying_;
+
+public:
+  template <typename ParseContext>
+  FMT_CONSTEXPR auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
+    return underlying_.parse(ctx);
+  }
+
   template <typename T, typename OutputIt>
   auto format(const T& value, basic_format_context<OutputIt, Char>& ctx) const
       -> OutputIt {
     auto buffer = basic_memory_buffer<Char>();
     format_value(buffer, value, ctx.locale());
-    return formatter<basic_string_view<Char>, Char>::format(
-        {buffer.data(), buffer.size()}, ctx);
+    return underlying_.format({buffer.data(), buffer.size()}, ctx);
```

Comment:
This change looks unnecessary.

Reply:
Oh, I missed your earlier comment. Since we support all string specs here I don't see any reason to disable set_debug_format, please revert.

Reply:
The reason I did it was... let's say we have this:
```cpp
struct Int {
    int i;
    friend std:ostream& operator<<(std::ostream& os, Int i) { return os << i.i; }
};

std::vector<Int> ints = {Int{1}, Int{2}, Int{3}};
fmt::print("ints={}\n", ints);
```

If you keep the debug format, this'll print as `["1", "2", "3"]` (because the library thinks it's formatting string_views).

Reply:
Good point but can we disable `set_debug_format` more explicitly e.g. keep the inheritance but delete this function in ostream formatter?

Reply:
When I did private inheritance, gcc 4.8 still picked it up anyway, but I didn't try `= delete;` which I agree is a more direct way of doing this. Will try and report back (or rather, if it works, I'll just change it. If it doesn't, then... I won't).

Reply:
Looks like that worked.